### ISSUE TITLE
Images inside `<script>` tags are broken

### ISSUE THREAD

**Issue by Tymvie** [futtta/autoptimize#382]
Hello,

Some plugins insert images into script tags, for example:
```html
<script>
var test = <?php echo wp_json_encode( '<img src="example.com">' ); ?>;
</script>
```

Autoptimize does not add backslashes before quotation marks. Therefore, the tag is broken:
```html
<script>
var test = "<img decoding="async" src=\"example.jpg\">";
</script>
```
Thanks for the great plugin

**Comment by futtta** [comment]
Thanks for reporting, I'll look into it, but I will likely (have to) stop AO's image optimization from acting on img tags in script tags, if not the chance of things breaking is too high anyway ..

just so I'm clear; this is when using the "image optimization" option on the "images" tab, right?

**Comment by Tymvie** [comment]
@futtta yes

**Comment by futtta** [comment]
Additionally; can you test if this also happens when "lazyload images" is on (it should not)?

**Comment by Tymvie** [comment]
@futtta yes, this does not happen when "lazyload images" is enabled

**Comment by futtta** [comment]
@tymvie I just committed a fix, can you [download the beta](https://github.com/futtta/autoptimize/archive/refs/heads/beta.zip) and test?

**Comment by Tymvie** [comment]
@futtta thank you, it works

**Comment by hrispm** [comment]
@futtta I can also confirm that this problem does not appear when the beta with the fix is installed.

**Comment by futtta** [comment]
just fyi; This was released as part of AO 3.0.4 earlier today :)

have a nice weekend and thank you for your assistance!

**Comment by Tymvie** [comment]
@futtta thank you, enjoy your weekend!
